### DISCUSSION TITLE
ComboBox component

### DISCUSSION THREAD

**Discussion by ezeamin** [saadeghi/daisyui#2226] [Ideas / feature requests]
Hi!

It will be really nice if daisyUI could implement a ComboBox component, that it's a select but with an integrated autocomplete search, as does [MUI](https://mui.com/material-ui/react-autocomplete/), [headless](https://headlessui.com/react/combobox), or [shadcn/ui](https://ui.shadcn.com/docs/components/combobox).

![image](https://github.com/saadeghi/daisyui/assets/82628932/0b9b516c-64d3-4304-977e-fc4fbc866309)

**Comment by saadeghi** [comment]
daisyUI is CSS. It's a plugin for Tailwind CSS and what you're asking for (the way mentioned libraries implemented the component) is not possible with CSS only. It needs JS.

However headless libraries like Headless UI or Radix are perfect to use alongside daisyUI because they only provide JS functionality without style and design decisions, while daisyUI only provides the style and design decisions.

Another option would be using them native HTML element `<dataset>`  
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist

**Reply by ezeamin** [reply]
Great, thanks!

**Reply by Zegoverno** [reply]
Hey @saadeghi would be awesome to use radix primitives with the style of daisyUI, but I am having some trouble to configure both.

**Comment by gevera** [comment]
Could we reopen this @saadeghi  ? 
IMHO dataset are OKish.

**Comment by waynesbrain** [comment]
Check it out - https://github.com/saadeghi/daisyui/discussions/4707

**Reply by waynesbrain** [reply]
@gevera if you don't use `Preact`, I bet ChatGPT can just convert it to whatever you want.

**Reply by gevera** [reply]
Far from Preact and JS world. I try to stay away from it as much as possible =)

**Reply by waynesbrain** [reply]
@gevera Welp, enjoy not having a combo box then

**Reply by saadeghi** [reply]
Please keep the conversation positive and friendly 💚
